src/terminal/parser/ft_fuzzer/fuzzing_directed.h

```diff
@@ -768,22 +771,27 @@ namespace fuzz
         // AddFuzzArrayEntry function can be used to add additional fuzz map 
         // entries without removing the existing map.  Returns E_INVALIDARG in the 
         // event that the total percentages add up to more than 100%.
+        [[nodiscard]]
         __inline HRESULT SetFuzzTypeMap(
             __in_ecount(cfte) const _fuzz_type_entry<_Type, _Args...> *rgfte,
             __in ULONG cfte) throw()
         {
             ClearFuzzTypeEntries();
-            for (ULONG i = 0; i < cfte; i++)
+
+            bool fInvalidEntry{};
+            for (ULONG i{}; i < cfte; ++i)

```

Comment:
I don't want to start a holy war here, but `i++` vs `++i`?

Reply:
There was a point in time or situation where `++i` was faster than `i++`. Not sure if it still holds.

Reply:
My intent nothing to do with speed; it's a personal habit of mine to avoid postfix wherever possible because it's too easy to write an off-by-one bug.  Before anyone asks, the ISO C++ guidelines are completely silent on this as far as I can tell; the only cursory reference I can find is to user-defined operators:

> Flag an unused return value from a user-defined non-defaulted postfix operator++ or operator-- function. Prefer using the prefix form instead.

Yes, historically, pre-increment/decrement was "faster" than post, because post creates a copy of the original value before incrementing, whereas pre doesn't.  However, in this case, I can see no actual difference in generated code so I imagine the compiler optimizes it away:

https://godbolt.org/z/1_dVhl

To put differently, I guess my reasoning would be that postfix form should be used only when the code explicitly wants the sequenced post-increment behavior.  Otherwise, code should always use prefix form.

The real problem with postfix form is that the author/reader suddenly have to be aware of sequencing points and [undefined behavior](https://herbsutter.com/2014/12/01/a-quick-poll-about-order-of-evaluation/
).  There's a great thread on stackoverflow the lists [many examples](https://stackoverflow.com/questions/949433/why-are-these-constructs-using-pre-and-post-increment-undefined-behavior).

However, this is definitely a question of project and personal style, and is not a correctness issue or a guidelines issues.  So I'm happy to adopt whatever the project deems as the preferred style.

Reply:
I am totally good with adopting `++i` as the preferred behavior given the advantages you list. 

It's just been my forever habit (since I learned to code) to do postfix. I could be retrained for a convincing benefit. And we could enforce it to some degree. Depends what the other folks think (@zadjii-msft, @adiviness, @DHowett-MSFT).

Reply:
I'm a fan of prefix by default, postfix where necessary, basically for the reasons listed above.

Reply:
prix fixe! prix fixe!

Reply:
Sounds like this decision should be documented. :)  Let's nip the "holy-war" before it starts.